### DISCUSSION TITLE
Display location issue

### DISCUSSION THREAD

**Discussion by nwt-q** [cpeditor/cpeditor#1418] [Q&A]
When I was using it, I placed the editor on a 4k screen and then removed the screen. I opened the compiler on my 2k screen, but it couldn't be located on the screen and was displayed outside the screen. I couldn't use the editor

**Comment by ouuan** [comment]
Now CP Editor remembers and restores the window size and position. This behavior was introduced in 4c69b87. I believe that it's either the window manager's or Qt's responsibility, not CP Editor's. I use i3wm so I'm not sure, though. Why do we need to do this @coder3101? I propose that we should remove this behavior.

**Comment by ouuan** [comment]
@nwt-q You can try removing the saved window geometry info in the settings (the line that looks like `geometry=@Rect(0 24 1920 1028)`). Tell me your CP Editor version and OS if you cannot find the settings file.

**Comment by coder3101** [comment]
Restoring window geometry is not an OS or WM responsibility. Our implementation for restore does not account for change in screen resolution or multi monitor setup.

So, we can adjust the window after the restore logic to make sure the window to a visible screen position.

We can achieve that with something like this:

```cpp

    void checkAndAdjustWindowGeometry()
    {
        // Get the screen geometry (for the primary screen)
        QRect screenGeometry = QApplication::primaryScreen()->availableGeometry();

        // Get the current window geometry
        QRect windowGeometry = geometry();

        // Check if the window is outside the screen bounds on any side
        if (windowGeometry.left() < screenGeometry.left()) {
            windowGeometry.moveLeft(screenGeometry.left());
        }
        if (windowGeometry.top() < screenGeometry.top()) {
            windowGeometry.moveTop(screenGeometry.top());
        }
        if (windowGeometry.right() > screenGeometry.right()) {
            windowGeometry.moveRight(screenGeometry.right());
        }
        if (windowGeometry.bottom() > screenGeometry.bottom()) {
            windowGeometry.moveBottom(screenGeometry.bottom());
        }

        // Apply the adjusted geometry if needed
        if (geometry() != windowGeometry) {
            setGeometry(windowGeometry);
        }
    }

```

**Reply by coder3101** [reply]
And we only adjust if window was restored outside visible area.

**Reply by nwt-q** [reply]
Thank you very much for resolving this issue. I also encountered a similar problem when dealing with Small project , which was solved by directly centering. CheckAndAdjusteWindowGeometry gave me great inspiration, and after verification, the effect was indeed significant. Thank you.

**Reply by coder3101** [reply]
Thanks for checking that above example works.  You need to wait for someone to implement this or you can open a PR with these changes and we will be happy to merge it.
